R/result.R

```diff
@@ -1,36 +1,343 @@
+# Merge two named vectors, overwriting identically-named elements of x1 with
+# that of x2.
+coalesce_vectors <- function(x1, x2) {
+  out <- x1
+  to_overwrite <- names(x2) %in% names(x1)
+  out[names(x2)[to_overwrite]] <- x2[to_overwrite]
+  out <- c(out, x2[!to_overwrite])
+  out
+}
+
+
+# Version of R6 with heritable static/class methods and attributes
+#
+# Elements in `static` can be called without instantiation, e.g. `Class$method()`.
+# Functions are evaluated in the environment of `Class`, so you can refer to `self`
+# (which is the class—not the instance—here) to create class methods.
```

Comment:
I wonder if some version of this might be useful in the README (or if that is too long in people's mind a vignette). This is a really good explanation and I think it could be useful to explain how/why one might approach using this.

Reply:
Added a pile more docs, including a version of the above